### PR TITLE
enhancement : Disable Deploy, Cluster Compare, Diff actions if cluster is no…

### PR DESCRIPTION
**Pull request by sriniva7**
This PR...

Issue number is #1238 

## How to test it

- Change the cluster configure in settings

## Screenshots

Below image show button are enabled when the cluster is available.
image

Below image show button are disabled when the cluster is available.
image

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/organisms/NavigatorPane/CompareButton.tsx [hunk 1]
```diff
@@ -7,13 +7,22 @@ import {SwapOutlined} from '@ant-design/icons';
...
 import WalkThrough from '@components/molecules/WalkThrough';
 
 export const CompareButton: React.FC<{children?: React.ReactNode}> = ({children}) => {
+  const isKubeConfigPathValid = useAppSelector(kubeConfigPathValidSelector);
+  useEffect(() => {
+    isCompareDisabled();
+  }, [isKubeConfigPathValid]);
   const dispatch = useAppDispatch();
-
+  const isCompareDisabled = () => {
```
Comment (topliceanurazvan) [inline src/components/organisms/NavigatorPane/CompareButton.tsx:21]: Here you could use `useMemo` instead, without needing a useEffect

**Review by topliceanurazvan — CHANGES_REQUESTED** [review]
I wouldn't disable the Compare button anymore as right now we are also allowing the user to compare local resources with helm/kustomize preview. Instead I would just disable the `Cluster` option from the dropdown ( screenshot ):

image

Also we are having hotkeys for quickly deploying and diffing, we should not allow the user to use them if the kubeconfig is not valid.

image


As another suggestions, when `Deploy/Diff` are disable, maybe we should show the tooltip with a different message like: `Your kubeconfig is not valid !`

**Comment by sriniva7** [comment]
> Great work so far 👍🏻 the only thing remaining is to not let the user use hotkeys. We can find it in `HotKeysHandler` component - do the action for `DIFF_RESOURCE` and `APPLY_SELECTION` only if the kubeconfig path is valid

Okay I will do that

File: src/components/organisms/HotKeysHandler/HotKeysHandler.tsx [hunk 2]
```diff
@@ -154,9 +156,10 @@ const HotKeysHandler = () => {
   useHotkeys(
     hotkeys.APPLY_SELECTION.key,
     () => {
+      if (!isKubeConfigPathValid) return;
```
Comment (topliceanurazvan) [inline src/components/organisms/HotKeysHandler/HotKeysHandler.tsx:159]:
```suggestion
      if (!isKubeConfigPathValid) {
        return;
      }

```

File: src/components/organisms/HotKeysHandler/HotKeysHandler.tsx [hunk 3]
```diff
@@ -168,9 +171,10 @@ const HotKeysHandler = () => {
   useHotkeys(
     hotkeys.DIFF_RESOURCE.key,
     () => {
+      if (!isKubeConfigPathValid) return;
```
Comment (topliceanurazvan) [inline src/components/organisms/HotKeysHandler/HotKeysHandler.tsx:174]:
```suggestion
      if (!isKubeConfigPathValid) {
        return;
      }

```

File: electron/app/menu.ts [hunk 4]
```diff
@@ -186,6 +188,7 @@ const fileMenu = (state: RootState, dispatch: MainDispatch): MenuItemConstructor
 };
 
 const editMenu = (state: RootState, dispatch: MainDispatch): MenuItemConstructorOptions => {
+  const isKubeConfigPathValid = useAppSelector(kubeConfigPathValidSelector);
```
Comment (topliceanurazvan) [inline electron/app/menu.ts:191]:
```suggestion
  const isKubeConfigPathValid = kubeConfigPathValidSelector(state);
```

File: electron/app/menu.ts [hunk 5]
```diff
@@ -231,6 +234,7 @@ const editMenu = (state: RootState, dispatch: MainDispatch): MenuItemConstructor
         accelerator: hotkeys.APPLY_SELECTION.key,
         enabled: Boolean(state.main.selectedResourceId),
```
Comment (topliceanurazvan) [inline electron/app/menu.ts:235]:
```suggestion
        enabled: Boolean(state.main.selectedResourceId) && Boolean(isKubeConfigPathValid),
```

We could just use the enabled property for making the option disabled if the kubeConfig path is not valid

File: electron/app/menu.ts [hunk 6]
```diff
@@ -239,6 +243,7 @@ const editMenu = (state: RootState, dispatch: MainDispatch): MenuItemConstructor
         accelerator: hotkeys.DIFF_RESOURCE.key,
         enabled: Boolean(state.main.selectedResourceId),
         click: () => {
+          if(!isKubeConfigPathValid)return;
```
Comment (topliceanurazvan) [inline electron/app/menu.ts:246]:
```suggestion
```

File: electron/app/menu.ts [hunk 7]
```diff
@@ -5,6 +5,8 @@ import hotkeys from '@constants/hotkeys';
 
 import {NewVersionCode, Project} from '@models/appconfig';
 import {RootState} from '@models/rootstate';
+import {useAppSelector} from '@redux/hooks';
+
```
Comment (topliceanurazvan) [inline electron/app/menu.ts:9]:
```suggestion
```

File: electron/app/menu.ts [hunk 8]
```diff
@@ -231,6 +234,7 @@ const editMenu = (state: RootState, dispatch: MainDispatch): MenuItemConstructor
         accelerator: hotkeys.APPLY_SELECTION.key,
         enabled: Boolean(state.main.selectedResourceId),
         click: () => {
+          if(!isKubeConfigPathValid)return;
```
Comment (topliceanurazvan) [inline electron/app/menu.ts:237]:
```suggestion
```

File: electron/app/menu.ts [hunk 9]
```diff
@@ -239,6 +243,7 @@ const editMenu = (state: RootState, dispatch: MainDispatch): MenuItemConstructor
         accelerator: hotkeys.DIFF_RESOURCE.key,
         enabled: Boolean(state.main.selectedResourceId),
```
Comment (topliceanurazvan) [inline electron/app/menu.ts:244]:
```suggestion
        enabled: Boolean(state.main.selectedResourceId) && Boolean(isKubeConfigPathValid),
```

Same as above

File: electron/app/menu.ts [hunk 10]
```diff
@@ -5,6 +5,8 @@ import hotkeys from '@constants/hotkeys';
 
 import {NewVersionCode, Project} from '@models/appconfig';
 import {RootState} from '@models/rootstate';
+import {useAppSelector} from '@redux/hooks';
```
Comment (topliceanurazvan) [inline electron/app/menu.ts:8]:
```suggestion
```

**Review by topliceanurazvan — CHANGES_REQUESTED** [review]

**Comment by topliceanurazvan** [comment]
Looks good 👍🏻 - left a couple of suggestions

**Review by topliceanurazvan — APPROVED** [review]
Great work @sriniva7 ! Thank you for your contribution 🙌🏻

**Pull request merged**